app/src/main/java/com/otaman/notes/viewmodel/AllNotesViewModel.kt

```diff
@@ -5,15 +5,27 @@ import androidx.lifecycle.*
 import com.otaman.notes.model.Note
 import com.otaman.notes.model.NoteRepository
 import com.otaman.notes.model.room.NoteDatabase
+import com.otaman.notes.view.MainActivity
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import java.util.*
 
 class AllNotesViewModel(application: Application): AndroidViewModel(application) {
-    val dao = NoteDatabase.getDatabase(application).getNoteDao()
-    val repository: NoteRepository = NoteRepository(dao)
+    private val dao = NoteDatabase.getDatabase(application).getNoteDao()
+    private val repository: NoteRepository = NoteRepository(dao)
+    private val _searchResults = MutableLiveData<List<Note>>()
+    val searchResults: LiveData<List<Note>> = _searchResults
     val allNotes: LiveData<List<Note>> = repository.getAllNotes()
 
     fun deleteNote(note: Note) = viewModelScope.launch(Dispatchers.IO) {
         repository.deleteNote(note)
     }
+
+    fun searchNote(searchNote: String) {
+        val result = allNotes.value?.filter { note ->
+            note.title.lowercase(Locale.ROOT).contains(searchNote) && note.content.lowercase(Locale.ROOT).contains(searchNote)
+        }.orEmpty()

```

Comment:
1. Does this logic work correctly when search query contains uppercase letters?
2. I would suggest to do filtering in a background thread

Reply:
1. Made filter logic to work correctly, replaced `&&` with `||` to have opportunity to search by `content`, not only by `title`
2. Made `searchNote()` to execute in a background thread